cranelift-codegen/meta/src/gen_legalizer.rs

```diff
@@ -401,14 +401,30 @@ fn gen_transform<'a>(
     // Guard the actual expansion by `predicate`.
     fmt.line("if predicate {");
     fmt.indent(|fmt| {
+        // If we are adding some blocks, we need to recall the original block, such that we can
+        // recompute it.
+        if !transform.block_pool.is_empty() {
+            fmt.line("let orig_ebb = pos.current_ebb().unwrap();");
```

Comment:
Maybe emit this unconditionally, and then always use the else case of `if transform.block_pool.is_empty() {` for `recompute_ebb`?

Reply:
Won't this cause recompute_ebb for each ebb containing a legalized instruction?

Reply:
I am not sure, as this implies keeping a variable with the value before other transformations as the one which is currently present would just pick the value without the need for extra stack space.

I do not think the rust compiler (LLVM) is capable of moving this code around and remove this extra variable.